dan_layer/core/src/workers/hotstuff_waiter.rs

```diff
@@ -1495,13 +1500,14 @@ where
 
         // all merkle proofs for the signers must be valid
         let validator_node_root = self.epoch_manager.get_validator_node_merkle_root(qc.epoch()).await?;
-        // TODO: Combine all validator merkle proofs before sending them
-        let mut verify_all = true;
-        for md in qc.validators_metadata() {
-            verify_all &= md
-                .merkle_proof
-                .verify(&validator_node_root, md.get_node_hash().to_vec());
-            if !verify_all {
+        if !qc.is_genesis() {
+            // Check the proof only for non-genesis blocks
+            let res = qc
+                .merged_proof()
+                .unwrap()
+                .verify_consume(&validator_node_root, qc.leave_hashes());
+            let res = res.unwrap();
```

Comment:
Can make a `BalancedBinaryMerkleProofError` variant in Hotstuff error
```suggestion
                .verify_consume(&validator_node_root, qc.leave_hashes())?;
```